pkg/limits/ingest_limits.go

```diff
@@ -123,6 +141,15 @@ func newMetrics(reg prometheus.Registerer) *metrics {
 type streamMetadata struct {
 	hash       uint64
 	lastSeenAt int64
+	totalSize  uint64
+	// Add a slice to track bytes per time interval for sliding window rate calculation
+	sizeBuckets []sizeBucket
```

Comment:
I think rate buckets is a good name?

```suggestion
	rateBuckets []sizeBucket
```